frontend/front/src/pages/Public/Components/LangPrefDropdown.js

```diff
@@ -0,0 +1,144 @@
+import React, { useState, useEffect } from "react";
+import { useLocation } from "react-router-dom";
+import { Box, Menu, MenuItem, Fade, Button, Typography } from "@mui/material";
+import LanguageIcon from "@mui/icons-material/Language";
+import KeyboardArrowDownIcon from "@mui/icons-material/KeyboardArrowDown";
+import { useTranslation } from "react-i18next";
+
+const LangPrefDropdown = () => {
+  const {
+    t,
+    i18n: { changeLanguage, language },
+  } = useTranslation();
+
+  const [anchorMore, setAnchorMore] = useState(null);
+  const [langDisplay, setLangDisplay] = useState("English");
+  const [currentLanguage, setCurrentLanguage] = useState(language);
+
+  const location = useLocation();
+
+  const langMap = {
+    "en-us": `🇺🇸 ${t("public.global-labels.locales.english")}`,
+    "ht-cr": `🇭🇹 ${t("public.global-labels.locales.creole")}`,
+    "pt-br": `🇧🇷 ${t("public.global-labels.locales.portuguese")}`,
+    "es-us": `🇪🇸 ${t("public.global-labels.locales.spanish")}`,
+  };
+
+  useEffect(() => {
+    // Get query params from current URL
+    const params = new URLSearchParams(window.location.search);
+    let queryLang = params.get("langPref");
+    // Check if current route is a 'public' route
+    const isPublicRoute = window.location.pathname.includes("public");
```

Comment:
Use react-router-dom's useLocation (https://v5.reactrouter.com/web/api/Hooks/uselocation) for this stuff (rather than the global window object), that way this useEffect will update if the location happens to change.

Reply:
Fixed it!